File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    var answer = 0;
    
    function gcd(a,b) {
        while(b!== 0) {
            let r = a%b;
            a=b;
            b=r;
        }
        return a;
    }
    
    function lcm(a,b){
        var target = (a*b) / gcd(a,b);
        return target;
    }
    
    var first = arr[0];
    for(var i=1;i<arr.length;i++){
        answer=lcm(first, arr[i]);
        first=answer;
}
   

    return answer;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    answer = 0
    
    # 두 수 최대 공약수 찾기
    def lcm(a,b) :
        tmp = 0
        while b != 0 :
            tmp = a
            a = b
            b = tmp % a
            
        return a
    
    if len(arr) <= 1 :
        return arr[0]
    cmp = arr[0] * arr[1] // lcm(arr[1], arr[0])
    for i in range(2, len(arr)) :
        cmp = cmp * arr[i] // lcm(max(cmp,arr[i]), min(cmp, arr[i]))
    
    
    return cmp